### PR TITLE
Enhancing Custom Theme Handling in front/css.php

### PR DESCRIPTION
**Pull request by ccailly**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

This Pull Request aims to improve the way custom themes are handled in the front/css.php file. Currently, when loading the relative path of the user's theme via front/css.php, it inadvertently exposes sensitive server structure information, which is undesirable.

To address this issue, this modification proposes the addition of a specific parameter is_custom_theme in front/css.php requests. When is_custom_theme is set to 1, GLPI will recognize that the request pertains to a custom theme and will load the theme directly from the correct directory, without exposing internal server paths. This enhancement enhances security by avoiding the disclosure of sensitive information.

Furthermore, this modification opens the door to a second improvement. In the event of an error while loading the custom theme (e.g., due to missing files or invalid SCSS), GLPI will automatically load the default theme. This ensures that users do not end up with a CSS-less GLPI in case of issues, and errors will be logged for later investigation.

### CODE REVIEW

File: front/css.php [hunk 1]
```diff
@@ -59,6 +60,17 @@
 // Ensure warnings will not break CSS output.
 ErrorHandler::getInstance()->disableOutput();
 
+// If a custom theme is requested, we need to get the real path of the theme
+if (isset($_GET['file']) && isset($_GET['is_custom_theme']) && $_GET['is_custom_theme']) {
+    $theme = ThemeManager::getInstance()->getTheme(preg_replace('/\.scss$/', '', $_GET['file']));
```
Comment (cedric-anne) [inline front/css.php:65]: Why adding the `.scss` extention to the `file` parameter to then remove it?

Comment (ccailly) [reply front/css.php]: The addition of the '.scss' was required to obtain a link with the css_path filter, but on reflection, this wasn't the right solution.

File: front/css.php [hunk 2]
```diff
@@ -59,6 +60,17 @@
...
 
+// If a custom theme is requested, we need to get the real path of the theme
+if (isset($_GET['file']) && isset($_GET['is_custom_theme']) && $_GET['is_custom_theme']) {
+    $theme = ThemeManager::getInstance()->getTheme($_GET['file']);
+
+    if (!$theme) {
+        $theme = ThemeManager::DEFAULT_THEME;
+    }
+
+    $_GET['file'] = $theme->getPath();
```
Comment (trasher) [inline front/css.php:71]: Won't work, `$theme` may be a simple string.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]
Code looks good to me (beside the issue already mentionned by @trasher).

One small addition, if you supply an invalid theme then GLPI UI is broken:

image

Maybe we could avoid the error by adding some try catch around the call to  `Html::compileScss`, with a fallback on the default theme ?

**Review by AdrienClairembault — APPROVED** [review]

File: front/css.php [hunk 3]
```diff
@@ -59,6 +60,17 @@
 // Ensure warnings will not break CSS output.
 ErrorHandler::getInstance()->disableOutput();
 
+// If a custom theme is requested, we need to get the real path of the theme
+if (isset($_GET['file']) && isset($_GET['is_custom_theme']) && $_GET['is_custom_theme']) {
+    $theme = ThemeManager::getInstance()->getTheme($_GET['file']);
+
+    if (!$theme) {
```
Comment (cedric-anne) [inline front/css.php:67]: Maybe a warning could be added. Without it, the administrator will not be able to understand why its custom theme is not rendered.

```suggestion
    if (!$theme) {
        trigger_error(sprintf('Unable to find theme `%s`.', $_GET['file']), E_USER_WARNING);
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**